test/jdk/java/lang/reflect/Proxy/LazyInitializationTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.lang.reflect.Proxy;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/*
+ * @test
+ * @bug 8285401
+ * @summary Avoid eager initialization of parameter types in proxy initialization
+ * @run testng LazyInitializationTest
+ */
+public final class LazyInitializationTest {
+    private static volatile boolean initialized = false;
+
+    interface Intf {
+        void m(Parameter parameter);
+    }
+
+    static class Parameter {
+        static {
+            initialized = true;
+        }
+    }
+
+    @Test
+    public void testLazyInitialization() {
+        Intf value = (Intf) Proxy.newProxyInstance(LazyInitializationTest.class.getClassLoader(),
+                new Class<?>[]{ Intf.class },
+                (proxy, method, args) -> null);
+        Assert.assertFalse(initialized, "parameter type initialized eagerly");
+
+        value.m(new Parameter());
+        Assert.assertTrue(initialized, "parameter type initialized after instantiation");
```

Comment:
> "parameter type initialized after instantiation"

Since this is the text that gets displayed/reported when the assertion fails, should this instead be "parameter type not initialized"?

Reply:
The rendered text for testng assert is `message expected: value actual: value`, so on a mismatch, it would print `parameter type initialized after instantiation expected: true actual: false`

Reply:
> so on a mismatch, it would print parameter type initialized after instantiation expected: true actual: false

That looks fine then. I hadn't looked closely at what testng prints and I had thought that on an assertion failure it would print only `AssertionError: parameter type initialized after instantiation` which wouldn't be accurate. The one you note above is more clear.